src/query.js

```diff
@@ -18,13 +19,15 @@ class Query {
    *
    * @param {DHT} dht - DHT instance
    * @param {Buffer} key
-   * @param {function(PeerId, function(Error, Object))} query - The query function to exectue
+   * @param {function(PeerId, Object, function(Error, Object))} query - The query function to exectue
+   * @param {function(Array<Object>)} aggregate - The aggregation function to combine multiple buckets
```

Comment:
As far as I understood, this should be `@param {function(Object, function(Error, Array<Object>))}`, right?

Reply:
Good catch! I think it actually ought to be just `@param {function(Object, function(Error, Object))}` since the result isn't necessarily an array; it's up to the caller to ensure that whatever the aggregate gives to its callback is what `query.run`'s callback expects.

Reply:
yes, I agree! 👍 

Change to that then:

```js
@param {function(Object, function(Error, Object))}
```

Reply:
New version should have this fixed.